File: Home-account/src/main/java/com/HomeFluent/Homeaccount/Services/UserService.java
package com.HomeFluent.Homeaccount.Services;

import java.util.List;

import com.HomeFluent.Homeaccount.dto.UserDto;
import com.HomeFluent.Homeaccount.entities.UserEntity;
import com.HomeFluent.Homeaccount.responseModels.UserRest;

import org.springframework.stereotype.Service;

@Service
public interface UserService {
    public UserRest getUserByName(String name);

    public UserRest getUserById(int id);

    public List<UserRest> getAlUsers();

    public UserRest createUser(UserDto user);

    public void deleteUser(String name);

    public UserRest updateUser(UserDto user);

}

File: Home-account/client/HomeFluentUI/src/public/scripts/Services/UserService.js
import httpcommon from "./http_common";
class UserService {
    createUser(user) {
        return httpcommon.post("/users/createUser", user);
    }
    getUsers() {
        return httpcommon.get();
    }
    getUserByName(name) {
        return httpcommon.get(name);
    }
    deleterByName(name) {
        return httpcommon.delete(name);
    }
    updateByName(name) {
        return httpcommon.delete(name);
    }
}

export default new UserService();